### PR TITLE
Properly handle printout to avoid stack smashing

### PR DESCRIPTION
**Pull request by aftersomemath**

### CODE REVIEW

File: src/MspFcComms.cpp [hunk 1]
```diff
@@ -78,12 +78,14 @@ namespace FcComms
...
         uint16_t raw_values[18];
         getRawRC(raw_values);
-        char RC_info[100];
-        int j = 0;
-        for (int i = 0 ; i < 18 ; i++) {
-            j+=sprintf(&RC_info[j], ", %d", raw_values[i]);
-        } //This is Levi's for loop
-        ROS_INFO(RC_info);
+
+        std::string RC_info;
```
Comment (amiller27) [inline src/MspFcComms.cpp]: Why are we using dynamic memory on the flight controller?  Also what's wrong with `snprintf`?

Comment (aftersomemath) [reply src/MspFcComms.cpp]: Good point, I just forgot about it.

Comment (amiller27) [reply src/MspFcComms.cpp]: :+1:

Comment (amiller27) [reply src/MspFcComms.cpp]: Good catch though.  Was this actually breaking things?

Comment (aftersomemath) [reply src/MspFcComms.cpp]: Yeah 100 bytes wasn't long enough anymore and the stack was being smashed. I wrote it originally so my bad.

File: src/MspFcComms.cpp [hunk 2]
```diff
@@ -78,11 +78,11 @@ namespace FcComms
     {
         uint16_t raw_values[18];
         getRawRC(raw_values);
-        char RC_info[100];
+        char RC_info[150];
         int j = 0;
         for (int i = 0 ; i < 18 ; i++) {
-            j+=sprintf(&RC_info[j], ", %d", raw_values[i]);
-        } //This is Levi's for loop
+            j+=snprintf(&RC_info[j], 150, ", %d", raw_values[i]);
```
Comment (amiller27) [inline src/MspFcComms.cpp]: Shouldn't this be `j >= 150 ? 0 : 150 - j` instead of just `150`?

Comment (aftersomemath) [reply src/MspFcComms.cpp]: yes

**Review by amiller27 — APPROVED** [review]

**Pull request merged**